feat: dfx deps subcommands (#3095)
dfinity/sdk: docs/concepts/canister-metadata.md

A string containing the wasm section content directly. Conflicts with "path".

## Canister Metadata Standard

> This section describes each metadata field (_in alphabetical order_)
> 

Reviewer: These were meant as PR comments
```suggestion
```

### `dfx:deps`

A list of `name:ID` pairs of direct dependencies separated by semicolon.

### `dfx:init`

A message to guide consumers how to initialize the canister.

### `dfx:wasm_hash` 

SHA256 hash of the wasm module located at `dfx:wasm_url`.

This section is optional.

In most cases, the wasm module at `dfx:wasm_url` will be the same as the on-chain wasm module. This means that dfx can read the state tree to obtain and verify the module hash.

In other cases, the wasm module at `dfx:wasm_url` is not the same as the on-chain wasm module. For example, the Internet Identity canister provides Development flavor to be integrated locally. In these cases, `dfx:wasm_hash` provides the expected hash, and dfx verifies the downloaded wasm against this.

### `dfx:wasm_url`

A URL to download canister wasm module which will be deployed locally.

## Requirements for "pull-ready" canisters

The "production" canister running on the mainnet should has following public metadata:

- `dfx:deps`
- `dfx:wasm_url`
- `dfx:wasm_hash` (optional)

> The canister wasm downloaded from `dfx:wasm_url` should has following metadata (public or private):

Reviewer: ```suggestion
A canister wasm downloaded from `dfx:wasm_url` should have the following metadata (public or private):
```

- `candid:service`
- `candid:args`
- `dfx:init`

> All the metadata sections with `dfx:` prefix must be set in `dfx.json` explicitly.

Reviewer: ```suggestion
All the metadata sections with the `dfx:` prefix must be set in `dfx.json` explicitly.
```

> Those two `candid:` metadata sections are handled by `dfx` during canister building.

Reviewer: ```suggestion
The two `candid:` metadata sections are handled by `dfx` during canister building.
```
